pkg/data/management/fleet_roles.go

```diff
@@ -0,0 +1,343 @@
+package management
+
+import (
+	"reflect"
+
+	"github.com/rancher/rancher/pkg/types/config"
+	v1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func ensureFleetRoles(management *config.ManagementContext) error {
+	fleetWorkspaceAdminRole := v1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "fleetworkspace-admin",
+		},
+		Rules: []v1.PolicyRule{
+			{
+				APIGroups: []string{
+					"fleet.cattle.io",
+				},
+				Resources: []string{
+					"gitrepos",
+				},
+				Verbs: []string{
+					"*",
+				},
+			},
+			{
+				APIGroups: []string{
+					"fleet.cattle.io",
+				},
+				Resources: []string{
+					"bundlenamespacemappings",
+				},
+				Verbs: []string{
+					"*",
+				},
+			},
+			{
+				APIGroups: []string{
+					"fleet.cattle.io",
+				},
+				Resources: []string{
+					"clusterregistrations",
+				},
+				Verbs: []string{
+					"*",
+				},
+			},
+			{
+				APIGroups: []string{
+					"fleet.cattle.io",
+				},
+				Resources: []string{
+					"bundledeployments",
+				},
+				Verbs: []string{
+					"*",
+				},
+			},
+			{
+				APIGroups: []string{
+					"fleet.cattle.io",
+				},
+				Resources: []string{
+					"clusters",
+				},
+				Verbs: []string{
+					"get",
+					"list",
+					"watch",
+				},
+			},
+			{
+				APIGroups: []string{
+					"fleet.cattle.io",
+				},
+				Resources: []string{
+					"contents",
+				},
+				Verbs: []string{
+					"*",
+				},
+			},
+			{
+				APIGroups: []string{
+					"fleet.cattle.io",
+				},
+				Resources: []string{
+					"bundles",
+				},
+				Verbs: []string{
+					"*",
+				},
+			},
+			{
+				APIGroups: []string{
+					"fleet.cattle.io",
+				},
+				Resources: []string{
+					"gitreporestrictions",
+				},
+				Verbs: []string{
+					"*",
+				},
+			},
+			{
+				APIGroups: []string{
+					"fleet.cattle.io",
+				},
+				Resources: []string{
+					"clusterregistrationtokens",
+				},
+				Verbs: []string{
+					"*",
+				},
+			},
+			{
+				APIGroups: []string{
+					"rbac.authorization.k8s.io",
+				},
+				Resources: []string{
+					"rolebindings",
+				},
+				Verbs: []string{
+					"*",
+				},
+			},
+		},
+	}
+
+	fleetWorkspaceMemberRole := v1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "fleetworkspace-member",
+		},
+		Rules: []v1.PolicyRule{
+			{
+				APIGroups: []string{
```

Comment:
member should be `*` on `gitrepos` and list,get,watch on all resources in the fleet apigroup